src/learn_move.c

```diff
@@ -105,15 +128,42 @@ void sub_8160664(u8 taskId)
 //   sub_81607EC();
 // 
 //   //gUnknown_0203BC38 ?
-//   sub_81D2824(*((u8*)&gUnknown_0203BC38 + 4));
+//   sub_81D2824(gUnknown_0203BC38[4]);
 //   sub_8161280();
 // 
-//   LoadSpriteSheet(/*FIXME &gUnknown_085CEBB0*/(void*)0x085CEBB0);
-//   LoadSpritePalette(/*FIXME &gUnknown_085CEBB8*/(void*)0x085CEBB8);
+//   LoadSpriteSheet(&gUnknown_085CEBB0);
+//   LoadSpritePalette(&gUnknown_085CEBB8);
+//   //LoadSpriteSheet(/*FIXME &gUnknown_085CEBB0*/(void*)0x085CEBB0);
+//   //LoadSpritePalette(/*FIXME &gUnknown_085CEBB8*/(void*)0x085CEBB8);
 // 
 //   data = ListMenuInit(&gMultiuseListMenuTemplate, *gUnknown_0203BC38, *((u8*)gUnknown_0203BC38 + 2));
 //   *(u8*)(gUnknown_0203BC34 + 274) = data;
 //   FillPalette(0, 0, 2);
 //   SetMainCallback2(/*FIXME sub_816082C*/(void*)0x0816082C);
 // }
-// */
+
+void sub_81607EC(void)
+{
+  ResetVramOamAndBgCntRegs();
+  ResetBgsAndClearDma3BusyFlags(0);
+  InitBgsFromTemplates(0, &gUnknown_085CEC28, 2);
+  ResetAllBgsCoordinates();
+  SetGpuReg(0, 0x1040);
+  ShowBg(0);
+  ShowBg(1);
+  SetGpuReg(0x50, 0);
+}
+void sub_816082C(void)
+{
+  //FIXME
+  //sub_8160868();
+  //void(*sub_8160868)(void) = (void (*)(void))0x08160868;
+  //sub_8160868();
+  __asm__("bl sub_8160868");
```

Comment:
No. Please dont do this. Just call it. It will generate the correct code.

Reply:
But the function is not yet decompiled. I will replace it.

Reply:
You can just extern it. It will work.